assets/src/stories-editor/components/block-mover/block-draggable.js

```diff
@@ -47,20 +95,85 @@ const BlockDraggable = ( { children, clientId, blockName, rootClientId, blockEle
 
 BlockDraggable.propTypes = {
 	index: PropTypes.number.isRequired,
-	rootClientId: PropTypes.string,
-	clientId: PropTypes.string,
-	blockElementId: PropTypes.string,
+	rootClientId: PropTypes.string.isRequired,
+	clientId: PropTypes.string.isRequired,
+	blockElementId: PropTypes.string.isRequired,
 	blockName: PropTypes.string,
 	children: PropTypes.any.isRequired,
 	onDragStart: PropTypes.func,
 	onDragEnd: PropTypes.func,
+	onNeighborDrop: PropTypes.func.isRequired,
+	getNeighborPageId: PropTypes.func.isRequired,
 };
 
-export default withSelect( ( select, { clientId } ) => {
-	const { getBlockIndex, getBlockRootClientId } = select( 'core/block-editor' );
-	const rootClientId = getBlockRootClientId( clientId );
-	return {
-		index: getBlockIndex( clientId, rootClientId ),
-		rootClientId,
-	};
-} )( BlockDraggable );
+export default compose(
+	withSelect( ( select, { clientId } ) => {
+		const { getBlockIndex, getBlockRootClientId, getBlockOrder, getBlock } = select( 'core/block-editor' );
+		const rootClientId = getBlockRootClientId( clientId );
+		return {
+			index: getBlockIndex( clientId, rootClientId ),
+			rootClientId,
+			block: getBlock( clientId ),
+			getBlockOrder,
+		};
+	} ),
+	withDispatch( ( dispatch, { getBlockOrder, rootClientId, clientId, block } ) => {
+		const { setCurrentPage } = dispatch( 'amp/story' );
+		const { selectBlock, removeBlock, insertBlock, updateBlockAttributes } = dispatch( 'core/block-editor' );
+
+		/**
+		 * Get id of neighbor page that is `offset` away from the current page.
+		 *
+		 * If no page exists in that direction, null will be returned.
+		 *
+		 * @param {number} offset  Integer specifying offset from current page - e.g. -2 on page 3 will return id of page 1. Page indices are zero-based.
+		 * @return {string} Returns id of target page or null if no page exists there.
+		 */
+		const getNeighborPageId = ( offset ) => {
+			const pages = getBlockOrder();
+			const currentPageIndex = pages.findIndex( ( i ) => i === rootClientId );
+			const newPageIndex = currentPageIndex + offset;
+			const isInsidePageCount = newPageIndex >= 0 && newPageIndex < pages.length;
+			const newPageId = pages[ newPageIndex ];
+
+			// Do we even have a neighbor in that direction?
+			if ( ! isInsidePageCount ) {
+				return null;
+			}
+
+			return newPageId;
+		};
+
+		/**
+		 * Drop this element on the page `offset` away from the current page (if possible). If dropped, update the position of the element on the new page.
+		 *
+		 * Currently this function removes the old element and creates a clone on the new page.
+		 *
+		 * @param {number} offset  Integer specifying offset from current page - e.g. -2 on page 3 will return id of page 1. Page indices are zero-based.
+		 * @param {Object} newAttributes  Object with attributes to update on element on new page.
+		 */
+		const onNeighborDrop = ( offset, newAttributes ) => {
```

Comment:
Please move this into it's own file / helpers. It would be extremely useful to reuse this code for #3330

Reply:
This PR https://github.com/ampproject/amp-wp/pull/3417 relays on this functions. They should be moved.

Reply:
Moved to a new helper hook in 1cde15c.